core/src/main/java/hudson/util/StreamTaskListener.java

```diff
@@ -128,7 +136,10 @@ public StreamTaskListener(File out, boolean append, Charset charset) throws IOEx
     }
 
     public StreamTaskListener(Writer w) throws IOException {
-        this(new WriterOutputStream(w));
+        // It's not possible to retrieve the charset that the writer is using;
+        // however, for all uses of this constructor, the writer is an instance
+        // of StringWriter, so it's okay to assume UTF-8.
+        this(new WriterOutputStream(w), StandardCharsets.UTF_8);
```

Comment:
The `WriterOutputStream` constructor needs to specify the encoding too, I think, or you will get junk.

Reply:
Thanks! Added.